### PR TITLE
fix: select all log source namespaces

### PR DESCRIPTION
**Pull request by csatib02**
https://github.com/kube-logging/telemetry-controller/pull/92 mistakenly removed the ability to select all namespaces, this PR adds that back.

To use this, specify the `logSourceNamespaceSelectors` field with an empty slice.
```yaml
apiVersion: telemetry.kube-logging.dev/v1alpha1
kind: Tenant
metadata:
  labels:
    collector: cluster
  name: tenant
spec:
  logSourceNamespaceSelectors: []
```

Also made the Collector CR's `ControlNamespace` field immutable.

### CODE REVIEW

**Comment by pepov** [comment]
as discussed: please add doc comments and make sure the behaviour is consistent with how subscriptionNamespaceSelectors behaves

**Review by OverOrion — APPROVED** [review]

File: charts/telemetry-controller/charts/opentelemetry-operator-0.74.3.tgz
Comment (pepov) [inline charts/telemetry-controller/charts/opentelemetry-operator-0.74.3.tgz:1]: this should not be part of the source code, only built and packaged at release time I beleive (should be solved in a separate pr)

File: api/telemetry/v1alpha1/tenant_types.go [hunk 1]
```diff
@@ -64,9 +64,12 @@ type RouteConfig struct {
 // TenantSpec defines the desired state of Tenant
 type TenantSpec struct {
 	SubscriptionNamespaceSelectors []metav1.LabelSelector `json:"subscriptionNamespaceSelectors,omitempty"`
-	LogSourceNamespaceSelectors    []metav1.LabelSelector `json:"logSourceNamespaceSelectors,omitempty"`
-	Transform                      Transform              `json:"transform,omitempty"`
-	RouteConfig                    RouteConfig            `json:"routeConfig,omitempty"`
+
+	// Determines the namespaces from which logs are collected by this tenant.
+	// If initialized with an empty list, logs from all namespaces are collected.
+	LogSourceNamespaceSelectors []metav1.LabelSelector `json:"logSourceNamespaceSelectors,omitempty"`
```
Comment (pepov) [inline api/telemetry/v1alpha1/tenant_types.go:70]: Can you please add comments for the other fields as well?
```suggestion
	// Determines the namespaces from which logs are collected by this tenant.
	// If initialized with an empty list, logs from all namespaces are collected.
	// If uninitialized, no logs are collected.
	LogSourceNamespaceSelectors []metav1.LabelSelector `json:"logSourceNamespaceSelectors,omitempty"`
```

File: charts/telemetry-controller/charts/opentelemetry-operator-0.74.3.tgz
Comment (csatib02) [reply charts/telemetry-controller/charts/opentelemetry-operator-0.74.3.tgz:1]: should we place it in the `.gitignore`, since it is always created when the chart is deployed?

Comment (csatib02) [reply charts/telemetry-controller/charts/opentelemetry-operator-0.74.3.tgz:1]: https://github.com/kube-logging/telemetry-controller/actions/runs/12200200336/job/34035810284?pr=110#step:9:6

When a non-optional dependency is needed, the package will always fail if it's not present it seems.
